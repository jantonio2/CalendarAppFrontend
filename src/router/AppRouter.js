import React, { useEffect } from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect
} from 'react-router-dom';
import { LoginScreen } from '../components/auth/LoginScreen';
import { CalendarScreen } from '../components/calendar/CalendarScreen';
import { useDispatch } from 'react-redux';
import { startChecking } from '../actions/auth';

export const AppRouter = () => {

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(startChecking());
  }, [dispatch]);

  return (
    <div>
      <Router>
        <div>
          <Switch>
            <Route exact path="/login" component={LoginScreen} />          
            <Route exact path="/" component={CalendarScreen} />

            <Redirect to="/" />          
          </Switch>
        </div>
      </Router>
    </div>
  );
};
