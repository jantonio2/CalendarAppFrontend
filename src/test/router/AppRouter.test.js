import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { AppRouter } from './../../router/AppRouter';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe('Pruebas en <AppRouter />', () => {
  
  test('debe de mostrar el Espere...', () => {
    const initState = {
      auth: {
        checking: true
      }
    };
    
    const store = mockStore( initState );

    const wrapper = mount(
      <Provider store={ store }>
        <AppRouter />
      </Provider>
    ); 
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h5').exists()).toBeTruthy();
  });
  
  test('debe de mostrar la ruta publica', () => {
    const initState = {
      auth: {
        checking: false,
      }
    };
    
    const store = mockStore( initState );
    
    const wrapper = mount(
      <Provider store={ store }>
        <AppRouter />
      </Provider>
    ); 
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.login-container').exists()).toBeTruthy();
  });

  test('debe de mostrar la ruta privada', () => {
    const initState = {
      auth: {
        checking: false,
        uid: 'TESTING',
        name: 'test'  
      },
      calendar: {
        events: []
      },
      ui: {
        openModal: false
      } 
    };
    
    const store = mockStore( initState );
    
    const wrapper = mount(
      <Provider store={ store }>
        <AppRouter />
      </Provider>
    ); 
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.calendar-screen').exists()).toBeTruthy();
  });

});
