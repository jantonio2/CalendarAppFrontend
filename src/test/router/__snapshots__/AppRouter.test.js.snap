// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el Espere... 1`] = `
<h5>
  Espere...
</h5>
`;
